chingu-voyages/v52-tier3-team-38
# Gary/feature/database

# Description
This PR includes the database connection and creation of data models. I also created a seed.js method to add fake data to the database. 

## Related to the following Jira/Trello issue 
Create a database to store resident contact information

- Insert link to a relevant task/issue in Jira/Trello
- https://trello.com/c/cM8UbjdC/40-create-a-database-to-store-resident-contact-information

## Main changes explained:
- Created dbConnect.js to connect to database.
- Created User and Appointment schemas under the models folder.
- Created seed.js to add some users to the database.
- Moved folders and files that were in the solarize folder out into the main directory.

## How to test:
1. check into current branch
2. do `npm install` and `npm run dev` to run this PR locally
3. Grab MONGODB_URL environment variable from the discord under "Environment Variables". Create a .env file to store it.
4. In your terminal, run "node utils/seed.js", the terminal should have logging showing that the connection to the database was successful, as well as showing the fake user data that was added to the database. 

## Note:
We discussed having 2 tables to separate the Residents and Admins and including roles on both tables. Upon implementing it, I condensed it into a single User table as I found the Admin table redundant since all admins would just have a 1 role and Residents have a role of 0. The second table just took up unnecessary space. I'm sure we can work with that without a problem. In terms of adding admins, this can be resolved with an API / manually adding it to the database since any further action is not in the requirements. I think it is best to keep it simple. I am happy to talk more about it on Discord or at the next stand-up, and of course make adjustments where seen fit.
